File: RedSheet.DbEntities/Models/vTargetRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTargetRecords", Schema = "dbo")]
	[Serializable]
	public partial class vTargetRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectNegotiationId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string TargetDetail { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TargetId { get; set; } 

	public vTargetRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-target-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTargetRecord {
    constructor(vTargetRecord?: vTargetRecord )  {
        let properties = [ "projectNegotiationId", "targetDetail", "targetId",];
        for (let property of properties)
            if (vTargetRecord && vTargetRecord[property])
                this[property] = vTargetRecord[property];
    }
 
	projectNegotiationId : number =   0 ;
 
    @required()
    @maxLength(1000)
	targetDetail : string =   undefined;
 
    @range(1,2147483647)
	targetId : number =   undefined;


}
